### ISSUE TITLE
No semi-colons false positive with enums

### ISSUE THREAD

**Issue by rocketraman** [pinterest/ktlint#2351]
(From https://github.com/detekt/detekt/issues/6615)

## Expected Behavior

I have a class like this:

```
enum class Foo {
  FOO,
  BAR,
  ;

  fun foo() = Unit
}
```

This should not report any error from `NoSemicolons`, because the `;` here is necessary.

My detekt configuration I have

```
  NoSemicolons:
    active: true
    autoCorrect: true
```

## Observed Behavior
The `NoSemicolons` rule is triggered with detekt 1.23.3. It is *not* triggered with detekt 1.22.0.

## Your Environment
* Version of ktlint used: 0.50.0 (from https://github.com/detekt/detekt/blob/v1.23.3/gradle/libs.versions.toml#L5)
* Relevant parts of the `.editorconfig` settings: N/A
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Detekt integration
* Version of Gradle used (if applicable): 8.4
* Operating System and version: `Linux edison 6.5.9-200.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Oct 25 20:40:49 UTC 2023 x86_64 GNU/Linux`

**Comment by paul-dingemans** [comment]
I can not reproduce this with ktlint CLI (`0.49.1`, '0.50.0`, '1.0.1`). I have no knowledge of Detekt. So I do not know how Detekts invokes ktlint. As long as it can not be reproduced with CLI, I can not be of help.
